spark_pipeline_framework/utilities/api_helper/http_request.py

```diff
@@ -159,9 +165,10 @@ def get_response(
         try:
             response.raise_for_status()
         except HTTPError as e:
-            raise Exception(
-                f"Request to {self.url} with arguments {json.dumps(arguments)} failed with {e.response.status_code}: {e.response.content}. Error= {e}"
-            ) from e
+            if self.raise_error:
+                raise Exception(
+                    f"Request to {self.url} with arguments {json.dumps(arguments)} failed with {e.response.status_code}: {e.response.content}. Error= {e}"
+                ) from e
```

Comment:
lets move this "if" above try block. We don't need to call raise_for_status if raise_error is False.